lib/spack/spack/fetch_strategy.py

```diff
@@ -290,6 +281,16 @@ def mirror_id(self):
         return os.path.sep.join(
             ['archive', self.digest[:2], self.digest])
 
+    @property
+    def curl(self):
+        if not self._curl:
+            try:
+                self._curl = which('curl', required=True)
+            except CommandNotFoundError as exc:
+                tty.error(str(exc) + "\n==> Spack will use urllib to fetch")
```

Comment:
It seems it is getting complicated (and potentially confusing) to fall back on urllib if curl isn't available. 

 I think we should revert this to the previous error message. If curl has been specified by the user but isn't available, the functionality should be the same as it is now.